lib/teiserver_web/live/account/profile/overview.ex

```diff
@@ -250,6 +285,38 @@ defmodule TeiserverWeb.Account.ProfileLive.Overview do
     |> assign(:profile_permissions, [])
   end
 
+  def handle_event("join", _params, %{assigns: assigns} = socket) do
+    user_id = assigns.user.id
+    current_user_id = assigns.current_user.id
+    lobby_id = assigns.client.lobby_id
+
+    with true <- client_connected(current_user_id),
+         {:ok, _} <- join_lobby(lobby_id, current_user_id) do
+      {:noreply, put_flash(socket, :success, "Lobby joined")}
+    else
+      {:error, reason} ->
+        {:noreply, put_flash(socket, :warning, reason)}
+    end
+  end
+
+  defp client_connected(user_id) do
+    client = Account.get_client_by_id(user_id)
+
+    if not is_nil(client) do
+      true
+    else
+      {:error, "Client is not connected"}
+    end
+  end
+
+  defp join_lobby(lobby_id, user_id) do
+    if :ok == Lobby.force_add_user_to_lobby(user_id, lobby_id) do
```

Comment:
Does this obey rules like user is banned from lobby and what not? I would have preferred to the use `add_user_to_battle`, but it has a `script_password` arg that I have no idea about

Reply:
I don't think it does, use `Battle.server_allows_join?` to check for that first.

This is how it's done in other places.
```ex
    if Battle.server_allows_join?(userid, lobby_id) == true do
      Battle.force_add_user_to_lobby(userid, lobby_id)
    end
```

Reply:
Thanks for the suggestion - I hadn't noticed it being used like that elsewhere. I added it [58d0bd8](https://github.com/beyond-all-reason/teiserver/pull/209/commits/58d0bd82fabc7461acdcfbf52c9606632561f892)